### ISSUE TITLE
cmd/go: shutdown crash output is unreadable

### ISSUE THREAD

**Issue by klauspost** [golang/go#46631]
### What version of Go are you using (`go version`)?

<pre>
λ gotip version
go version devel go1.17-5542c10fbf Fri Jun 4 03:07:33 2021 +0000 windows/amd64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
λ gotip env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=d:\temp\gocache
set GOENV=C:\Users\klaus\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=e:\gopath\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=e:\gopath
set GOPRIVATE=
set GOPROXY=https://goproxy.io
set GOROOT=C:\Users\klaus\sdk\gotip
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=C:\Users\klaus\sdk\gotip\pkg\tool\windows_amd64
set GOVCS=
set GOVERSION=devel go1.17-5542c10fbf Fri Jun 4 03:07:33 2021 +0000
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=NUL
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=d:\temp\wintemp\go-build102784969=/tmp/go-build -gno-record-gcc-switches
</pre></details>

### What did you do?

Simple fuzz test:

```Go
func FuzzCrash(f *testing.F) {
	f.Cleanup(func() {
		panic("error")
	})
	f.Fuzz(func(t *testing.T, data []byte) {
		time.Sleep(time.Second)
	})
}
```

Run `gotip test -fuzz=FuzzCrash -test.run=None -parallel=16`.  Press ctrl+c which caused a crash.

### What did you expect to see?

Readable output.

### What did you see instead?

Copy of output: https://gist.github.com/klauspost/a03eaeda5aa172c0a5990f91c6e7b4a9

(Also we obviously didn't call "f.Fail", the CleanUp function probably did).

**Comment by katiehockman** [comment]
I just repro'd this locally and can confirm that something is going wrong here. Will look into it further.

**Comment by katiehockman** [comment]
It looks like every worker is calling the `Cleanup` method, and panicking in a way that isn't being caught/logged properly. That very large block of text appears to be a set of stack traces that came from each worker panicking during Cleanup (in your example, you had 16 workers, so that's going to be a lot of output).

**Comment by jayconrod** [comment]
I looked into this a bit today. Can confirm, the output is garbled because the worker processes share stdout and stderr. I set it up that way initially to help with debugging, but we should fix it and do it right.

`testing` uses stdout for test control messages (`--- PASS` and so on with `-v`) and logging (via `TB.Log`). The coordinator process should probably capture stdout, then only forward log lines onto its own stdout.

`testing` doesn't usually write to stderr (except when parsing its own flags), so anything written there should come from the fuzz target or an unrecovered panic (which is the case here). The coordinator should capture stderr from each worker, then print that or return an error when a worker terminates unexpectedly.

If all the workers panic, the output is going to be overwhelming, even if the coordinator prints each worker's output separately. We may just want to just print output from the first failing worker.
